binaryfusefilter.go

```diff
@@ -114,13 +114,8 @@ func PopulateBinaryFuse8(keys []uint64) (*BinaryFuse8, error) {
 		iterations += 1
 		if iterations > MaxIterations {
 			// The probability of this happening is lower than the
-			// the cosmic-ray probability (i.e., a cosmic ray corrupts your system),
-			// but if it happens, we just fill the fingerprint with ones which
-			// will flag all possible keys as 'possible', ensuring a correct result.
-			for i := 0; i < len(filter.Fingerprints); i++ {
-				filter.Fingerprints[i] = ^uint8(0)
-			}
-			return filter, nil
+			// the cosmic-ray probability (i.e., a cosmic ray corrupts your system).
+			return nil, errors.New("too many iterations")
```

Comment:
would be better to define this as an exported error and return the error instead.
i.e.
```golang
var ErrTooManyIterations = errors.New("too many iterations")
...
return nil, ErrTooManyIterations
```
when testing this, you can you the explicit exported variable instead of comparing strings.

Reply:
Good point. Want to issue a pull request?

Reply:
my computer isn't set up for that at the moment.. but maybe you could roll this onto the next set of changes you'll make ?